packages/sync/src/links/OfflineQueueLink.ts

```diff
@@ -62,12 +63,28 @@ export class OfflineQueueLink extends ApolloLink {
     this.operationFilter = filter;
   }
 
-  public open() {
+  public async open() {
```

Comment:
I believe that with this change and entire thing being async now we can get rid or refactor `OfflineRestoreHandler`.